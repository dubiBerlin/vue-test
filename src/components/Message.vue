<template>
  <li style="margin-top: 10px" class="message">
    <span>{{ message }}</span>
  </li>
</template>
<script>
import axios from "axios";

export default {
  props: ["message"],
  created() {
    axios
      .get(
        "https://8174d70e-822b-40f0-ba26-9033f398d579.mock.pstmn.io/hello/world",
        {
          params: {
            request: 1,
            from: "2019-08-23",
            till: "2019-08-24"
          }
        }
      )
      .then(response => console.log("RESPONSE: ", response.data));
  }
};
</script>
