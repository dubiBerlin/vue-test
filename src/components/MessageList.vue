<template>
  <ul>
    <li v-for="message in messages" v-bind:key="message">{{ message }}</li>
  </ul>
</template>
<script>
export default {
  name: "list",
  props: ["messages"]
};
</script>
