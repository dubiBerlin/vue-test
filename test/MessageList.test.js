import { mount } from '@vue/test-utils'
import MessageList from '../src/components/MessageList'
import Message from '../src/components/Message.vue'

describe('MessageList.test.js', () => {
  let cmp
  beforeEach(() => {
    cmp = mount(MessageList, {
      // Be aware that props is overridden using `propsData`
      propsData: {
        messages: ['Cat']
      }
    })
  })

  it('is a MessageList component', () => {
    expect(cmp.is(MessageList)).toBe(true)
    expect(cmp.is('ul')).toBe(true)
  })

  it('has .message class ', () => {
    expect(cmp.find('.message').element).toBeInstanceOf(HTMLElement)
  })

  it('has received ["Cat"] as the message property', () => {
    expect(cmp.vm.messages).toEqual(['Cat'])
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
  it('contains a Message component', () => {
    expect(cmp.contains(Message)).toBe(true)
    // Or with CSS selector
    expect(cmp.contains('.message')).toBe(true)
  })

  it('Both, MessageList and Message are vue instances', () => {
    expect(cmp.isVueInstance()).toBe(true)
    expect(cmp.find(Message).isVueInstance()).toBe(true)
  })
})
