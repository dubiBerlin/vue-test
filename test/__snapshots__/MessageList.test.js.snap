// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.test.js has the expected html structure 1`] = `
<ul>
  <li
    class="message"
  >
    <span>
      Cat
    </span>
  </li>
</ul>
`;
